components/support/nimbus-fml/src/parser.rs

```diff
@@ -443,12 +485,11 @@ fn merge_map<T: Clone>(
 fn merge_import_block(a: &ImportBlock, b: &ImportBlock) -> Result<ImportBlock> {
     let mut block = a.clone();
 
-    for (id, defaults) in &b.features {
-        let mut defaults = defaults.clone();
+    for (id, additions) in &b.features {
         if let Some(existing) = block.features.get_mut(id) {
-            existing.append(&mut defaults);
+            existing.merge(additions);
         } else {
-            block.features.insert(id.clone(), defaults.clone());
+            block.features.insert(id.clone(), additions.clone());
         }
```

Comment:
This should use the Entry API, which avoids double-hashing of the key in the empty case.

```rust
block.features
    .entry(id.clone())
    .and_modify(|existing| existing.merge(additions))
    .or_insert_with(|| additions.clone());
```

Reply:
Nice!